server/consumer.go

```diff
@@ -749,6 +749,16 @@ func (o *consumer) checkQueueInterest() {
 	}
 }
 
+// clears our node if we have one. When we scale down to 1.
+func (o *consumer) clearNode() {
+	o.mu.Lock()
+	if o.node != nil {
+		o.node.Delete()
+		o.node = nil
+	}
+	defer o.mu.Unlock()
```

Comment:
Weird to have the defer here. Either you have this after lock, or remove defer.

Reply:
Note also that although that we often check for o.node != nil, I did not find that we were setting it to nil, except now.

Reply:
Will fix.

We set it in setConsumerAssignment.